lib/utils/html-fn.html

```diff
@@ -0,0 +1,61 @@
+<!--
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="boot.html">
+<script>
+  (function() {
+    'use strict';
+
+    /**
+     * @param {*} value Object to stringify into HTML
+     * @return {string} HTML stringified form of `obj`
+     */
+    function htmlValue(value) {
+      if (value instanceof HTMLElement) {
+        return /** @type {!HTMLElement} */(value).innerHTML;
+      } else if (value instanceof Node) {
+        return /** @type {!Node} */(value).textContent;
+      } else {
+        return value.toString();
+      }
+    }
+
+    /**
+     * Small tagged template string function to facilitate building a `<template>` element.
+     * Polymer.html will automatically compose HTML elements in `${}` blocks, allowing for easy
+     * composition of superclass templates, or partial templates.
+     *
+     * Example:
+     *
+     *   static get template() {
+     *     return Polymer.html`
+     *       <style>:host{ content:"..." }</style>
+     *       <div class="shadowed">${this.partialTemplate}</div>
+     *       ${super.template}
+     *     `;
+     *   }
+     *   static get partialTemplate() { return Polymer.html`<span>Partial!</span>`; }
+     *
+     * @memberof Polymer
+     * @param {!Array<string>} strings Constant parts of tagged template literal
+     * @param {!Array<*>} values Variable parts of tagged template literal
+     * @return {!HTMLTemplateElement} Constructed HTMLTemplateElement
+     */
+    Polymer.html = function html(strings, ...values) {
+      /** @type {!Array<string>} */
+      const parts = [strings[0]];
```

Comment:
🏌️: you can get the interpolation with:

```js
const htmlString = values.reduce((r, v, i) => r + htmlValue(v) + strings[i+1], strings[0]);
```

Reply:
wouldn't it be more efficient to do the concatenation at the end?

Reply:
Apparently string concatenation is so optimized that it's faster than join. There's a rope version of a string in V8, so that kind of makes sense.

Reply:
SGTM